nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.message.SchemaStore;
+import org.nuxeo.runtime.RuntimeServiceException;
+
+/**
+ * @since 10.2
+ */
+public class AvroServiceImpl
+        extends SchemaStore.Cache
+        implements AvroService {
+
+    private static final AvroMapper<Object, Object> NULL = new AvroMapper<Object, Object>(null) {
+
+        @Override
+        public Object fromAvro(Schema schema, Object input) {
+            return null;
+        }
+
+        @Override
+        public GenericRecord toAvro(Schema schema, Object input) {
+            return null;
+        }
+
+    };
+
+    protected final Map<Class<?>, Class<AvroSchemaFactory<?>>> factories;
+
+    protected final List<AvroReplacementDescriptor> replacements;
+
+    protected Map<Class<?>, AvroMapper<?, ?>> mappers;
+
+    public AvroServiceImpl(Collection<AvroReplacementDescriptor> replacements,
+            Map<Class<?>, Class<AvroSchemaFactory<?>>> factories) {
+        this.replacements = new ArrayList<>(replacements);
+        this.factories = new HashMap<>(factories);
+        this.replacements.sort(null);
```

Comment:
What do you think about `Comparator.naturalOrder()` instead of `null` ?

Reply:
the javadoc says ;
Parameters: c the Comparator used to compare list elements. A null value indicates that the elements' natural ordering should be used